backend/typescript/services/implementations/authService.ts

```diff
@@ -0,0 +1,81 @@
+import * as firebaseAdmin from "firebase-admin";
+
+import UserService from "./userService";
+import IAuthService from "../interfaces/authService";
+import IUserService from "../interfaces/userService";
+import { Token, Role } from "../../types";
+import FirebaseRestClient from "../../utilities/firebaseRestClient";
+import Logger from "../../utilities/logger";
+
+class AuthService implements IAuthService {
+  userService: IUserService;
+
+  constructor() {
+    this.userService = new UserService();
+  }
+
+  /* eslint-disable class-methods-use-this */
+  async generateToken(email: string, password: string): Promise<Token> {
+    try {
+      return await FirebaseRestClient.signInWithPassword(email, password);
```

Comment:
Do you need an `await` if you're returning?

Reply:
I think it's needed in a try-catch so that the catch block will actually execute if the promise is rejected